src/test/java/bio/terra/service/snapshot/SnapshotTest.java

```diff
@@ -285,4 +294,63 @@ public void snapshotByFullViewHappyPathTest() throws Exception {
         assertEquals("new snapshot has been created", snapshot.getName(), requestModel.getName());
         assertEquals("all 5 relationships come through", snapshot.getRelationships().size(), 5);
     }
+
+    @Test
+    public void snapshotACLTest() throws Exception {
+        DatasetModel dataset = dataRepoFixtures.getDataset(steward(), datasetId);
+
+        String datasetName = dataset.getName();
+
+        logger.info("---- Dataset ACLs before snapshot create-----");
+        int datasetAclCount = fetchSourceDatasetAcls(datasetName);
+
+        //-------------------Create Snapshot----------------
+        SnapshotRequestModel requestModel =
+            jsonLoader.loadObject("ingest-test-snapshot-fullviews.json", SnapshotRequestModel.class);
+        // swap in the correct dataset name (with the id at the end)
+        requestModel.getContents().get(0).setDatasetName(datasetName);
+        SnapshotSummaryModel snapshotSummary =
+            dataRepoFixtures.createSnapshotWithRequest(steward(),
+                datasetName,
+                profileId,
+                requestModel);
+        TimeUnit.SECONDS.sleep(10);
+        createdSnapshotIds.add(snapshotSummary.getId());
+        SnapshotModel snapshot = dataRepoFixtures.getSnapshot(steward(), snapshotSummary.getId());
+        assertEquals("new snapshot has been created", snapshot.getName(), requestModel.getName());
+        assertEquals("all 5 relationships come through", snapshot.getRelationships().size(), 5);
+
+        // fetch ACLs
+        logger.info("---- Dataset ACLs after snapshot create-----");
+        int datasetPlusSnapshotCount = fetchSourceDatasetAcls(datasetName);
+        assertThat("There should be more ACLs on the dataset after snapshot create",
+            datasetPlusSnapshotCount, greaterThan(datasetAclCount));
+
+        //-----------delete snapshot------------
+        dataRepoFixtures.deleteSnapshot(steward(), snapshotSummary.getId());
+
+        // Give ACLs a second to update in big query
+        TimeUnit.SECONDS.sleep(10);
```

Comment:
I thought it could take up to 7 minutes for the ACLs to update. Could that cause this test to be flakey?

Reply:
That's a great point. I've add a retry with exponential backoff to wait up to 7 minutes for the ACLs to update. Hopefully this should take care of any flakiness.